lib/Core/Site/Pagination/Pagerfanta/ContentSearchHitAdapter.php

```diff
@@ -49,10 +54,21 @@ public function getNbResults()
             return $this->nbResults;
         }
 
-        $countQuery = clone $this->query;
-        $countQuery->limit = 0;
+        return $this->nbResults = $this->getSearchResult()->totalCount;
+    }
+
+    /**
+     * Returns the facets of the results.
+     *
+     * @return \eZ\Publish\API\Repository\Values\Content\Search\Facet[] The facets of the results
+     */
+    public function getFacets()
```

Comment:
Looks good, but there is no reason for having separate methods for facets and total count. They can both be set from a single search result, not doing that means executing extra query. So maybe something like:

```php
private $areLimitZeroPropertiesInitialized = false;

private function initializeLimitZeroProperties()
{
    // Perform search with limit 0

    $this->facets = ...
    $this->totalCount = ...

    $this->areLimitZeroPropertiesInitialized = true;
}

public function getFacets()
{
    if (!$areLimitZeroPropertiesInitialized) {
        $this->initializeLimitZeroProperties();
    }

    return $this->facets;
}

...
```

Reply:
The` searchResult` is cached.  A new search is only performed if `$this->searchResult === null`.  `$this->searchResult` will be populated by whichever method performs the query first so no extra query would be performed.  If you prefer you example that is fine too, I can change it.

Reply:
No need, that's good :) Sorry, I didn't look carefully enough.